addons/mrp/models/mrp_production.py

```diff
@@ -407,7 +411,11 @@ def action_view_mo_delivery(self):
             else:
                 action['views'] = form_view
             action['res_id'] = pickings.id
-        action['context'] = dict(action.get('context', {}), create=False, default_origin=self.name)
+        action['context'] = dict(
+            safe_eval(action['context'].strip(), eval_context),
```

Comment:
I think that this is not the right evaluation context, see https://github.com/odoo/odoo/blob/14.0/odoo/addons/base/models/ir_actions.py#L245-L250
The stuff that you had to add (`allowed_company_ids`) is way too specific and fragile.